### PR TITLE
Removes unused createStaticConfigProvider

### PR DESCRIPTION
**Pull request by mathetake**
Commit Message: Removes unused createStaticConfigProvider
Additional Description:
Previously, a variant of createStaticConfigProvider taking `Protobuf::Message` 
as a first param was not used, and that was one reason for the low test coverage 
of envoy/config package. This removes it, and potentially allows the package 
to be removed from the coverage exception.

Risk Level: low
Testing: done
Docs Changes: n/a
Release Notes: n/a
Platform Specific Features: n/a

### CODE REVIEW

File: test/common/config/config_provider_impl_test.cc [hunk 1]
```diff
@@ -195,19 +195,11 @@ class DummyConfigProviderManager : public ConfigProviderManagerImplBase {
 
   // Envoy::Config::ConfigProviderManager
   ConfigProviderPtr
-  createStaticConfigProvider(const Protobuf::Message& config_proto,
+  createStaticConfigProvider(std::vector<std::unique_ptr<const Protobuf::Message>>&& configs,
```
Comment (adisuissa) [inline test/common/config/config_provider_impl_test.cc:198]: nit: can the type be replaced with `ProtobufTypes::ConstMessagePtrVector`?

**Review by adisuissa — COMMENTED** [review]
Thanks overall LGTM, modulo small nit.

File: test/common/config/config_provider_impl_test.cc (hunk 1 above)
Comment (mathetake) [reply test/common/config/config_provider_impl_test.cc]: thanks for the good catch, done in https://github.com/envoyproxy/envoy/pull/34985/commits/f465f61c8aac890bd0883524682dbfb6c738cac7

File: test/common/config/config_provider_impl_test.cc [hunk 2]
```diff
@@ -195,19 +195,11 @@ class DummyConfigProviderManager : public ConfigProviderManagerImplBase {
...
-        dynamic_cast<const test::common::config::DummyConfig&>(config_proto), factory_context,
-        *this);
-  }
-  ConfigProviderPtr
-  createStaticConfigProvider(std::vector<std::unique_ptr<const Protobuf::Message>>&&,
-                             Server::Configuration::ServerFactoryContext&,
                              const OptionalArg&) override {
-    ASSERT(false, "this provider does not expect multiple config protos");
-    return nullptr;
+    auto config = dynamic_cast<const test::common::config::DummyConfig&>(*configs[0]);
```
Comment (adisuissa) [inline test/common/config/config_provider_impl_test.cc:201]: nit: add `ASSERT(configs.size() == 1);` and a comment saying that currently the provide does not expect multiple config protos.

**Review by adisuissa — COMMENTED** [review]
Thanks!
I'm sorry but I missed something minor previously.
After this, it should be good to go.
Assigning Harvey for a final review and merge.
/assign @htuch

File: test/common/config/config_provider_impl_test.cc (hunk 2 above)
Comment (mathetake) [reply test/common/config/config_provider_impl_test.cc:204]: https://github.com/envoyproxy/envoy/pull/34985/commits/3f86042de1e00152c52ea84b0fa2eaaf9d021a51 👍

**Review by htuch — APPROVED** [review]
LGTM, thanks!

**Pull request merged**